src/sentry/integrations/opsgenie/integration.py

```diff
@@ -121,9 +121,13 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:  # type:ignore
 
 
 class OpsgenieIntegration(IntegrationInstallation):
-    def get_client(self) -> Any:
+    def get_client(self, integration_key: str) -> Any:  # type: ignore
         org_integration_id = self.org_integration.id if self.org_integration else None
-        return OpsgenieClient(integration=self.model, org_integration_id=org_integration_id)
+        return OpsgenieClient(
+            integration=self.model,
+            integration_key=integration_key,
+            org_integration_id=org_integration_id,
+        )
```

Comment:
I don't think it's necessary right now, but it may make more sense to have two different clients. One for requests using team integration keys and the other for account integration keys. We can talk more about this offline, but I'm just seeing areas where someone extending Opsgenie without the context we have could pass in the wrong key and fail a bunch of calls on the client with 403 errors.

Reply:
I've taken a look, and I believe that the setup client is the only one that uses the account integration key. But I agree that we should discuss safeguards against passing the wrong key!